### PR TITLE
Replace logging related dependency with more actively maintained package and update dependencies.

### PR DESCRIPTION
**Pull request by YooSunYoung**
# Replace custom log handler with ``rich`` log handler.
`colorama` was an additional dependency for logging feature, but it seems like they are not actively maintained recently.
So I replaced stream handler with `rich` log handler instead so that we don't need to have `colorama` as a dependency anymore.
While replacing ``colorama`` with ``rich``, I removed ``HeaderMixin`` which had an interface to emit ``header`` of the log columns, but for now I removed them as well.
It still uses `HeaderColumn`s, so this functionality can be added later easily if needed.

# Update dependencies
``scippneutron`` was added as a nightly dependency.
After all changes, all dependency recipes were updated as well.

# Nightly dependencies...?
So all dependencies that are required for default testing are now in `base.in` or `nightly.in`.

But `nightly.in` don't have all dependencies in `extra.in` for non-default testings.
I was wondering if this shouldn't be the case at all or  if it's something we have to fix...?

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
> But `nightly.in` don't have all dependencies in `extra.in` for non-default testings.
> I was wondering if this shouldn't be the case at all or if it's something we have to fix...?

I don't think that can be done automatically, since `nightly-in` is auto-generated. Can you include those deps differently, somehow?

File: requirements/extra.in [hunk 1]
```diff
@@ -1,7 +1,5 @@
 confluent-kafka
-colorama
+ess-streaming-data-types
 scipp
 scippneutron
 plopp
-h5py
-ess-streaming-data-types
```
Comment (SimonHeybrock) [inline requirements/extra.in:7]: Given that only ESS will use Beamlime, is there any point in making these optional/extra? Might be simpler to just put these into `pyproject.toml`, at least until someone actually wants to use this for something else (which is really unlikely).

Comment (YooSunYoung) [reply requirements/extra.in]: I thought it should be done as they become actual dependencies of the package,
but and then I thought it doesn't really matter... so I added them...! Thanks!

File: pyproject.toml [hunk 2]
```diff
@@ -30,7 +30,8 @@ requires-python = ">=3.9"
 # Run 'tox -e deps' after making changes here. This will update requirement files.
 # Make sure to list one dependency per line.
 dependencies = [
-  "numpy"
+  "numpy",
+  "rich"
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply pyproject.toml]: Can't see the change you, did you commit it?

[the remark this answers is not in the document]
Comment (YooSunYoung) [reply pyproject.toml]: > Can't see the change you, did you commit it?

Sorry, I just did...!

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**